employee&managment/service.js

```diff
@@ -0,0 +1,69 @@
+// service.js
+const { Employee, Department } = require('./models');
+
+class EmployeeService {
+  async addEmployee({ firstName, lastName, salary, departmentName, hireDate }) {
+    let department = await Department.findOne({ where: { name: departmentName } });
+    if (!department) {
+      department = await Department.create({ name: departmentName });
+    }
+    const employee = await Employee.create({
+      firstName,
+      lastName,
+      salary,
+      hireDate,
+      departmentId: department.id,
+    });
+    return employee;
+  }
```

Comment:
<div>


<div id="suggestion">
<div id="issue"><b>Consider adding input validation for employee</b></div>
<div id="fix">

The `addEmployee` method could benefit from input validation. Consider validating `salary` is positive and `hireDate` is not in the future before creating the employee record.
</div>
<details>
<summary>
<b>Code suggestion</b>
</summary>
<div id="code">

```
 @@ -5,6 +5,13 @@
    async addEmployee({ firstName, lastName, salary, departmentName, hireDate }) {
 +    if (salary <= 0) {
 +      throw new Error('Salary must be greater than 0');
 +    }
 +
 +    if (new Date(hireDate) > new Date()) {
 +      throw new Error('Hire date cannot be in the future');
 +    }
 +
      let department = await Department.findOne({ where: { name: departmentName } });
```
</div>
</details>
</div>


<small><i>Code Review Run <a href=https://github.com/rishabhbohra3/Bito-CRA-Issues-Dataset/pull/3#issuecomment-2510872065>#7d3d92</a></i></small>
</div>

Is this a valid issue, or was it incorrectly flagged by the Agent?
- [ ] it was incorrectly flagged